In [ ]:
from langchain_openai import OpenAIEmbeddings
import os
from dotenv import load_dotenv
load_dotenv()   
api_key = os.getenv("OPENAI_API_KEY")

In [12]:
embedding_model = OpenAIEmbeddings(api_key=api_key)

In [ ]:
text = "I am PremKumar, you tutor and mentor for RAG and AI"
vector = embedding_model.embed_query(text)

In [ ]:
# Let's see what it looks like (it will be a list of floating point numbers)
print(f"Vector length: {len(vector)}")
print(f"First 5 numbers: {vector[:5]}")

Vector length: 1536
First 5 numbers: [0.0004620793624781072, -0.0030264039523899555, -0.010309120640158653, -0.031093191355466843, -0.0004767622740473598]


In [21]:
# Let's see what it looks like (it will be a list of floating point numbers)
text = "I am PremKumar, you tutor and mentor for RAG and AI"
vector = embedding_model.embed_query(text)
print(f"Vector length: {len(vector)}")
print(f"First 5 numbers: {vector}")

Vector length: 1536
First 5 numbers: [0.00520513579249382, -0.02007550187408924, -0.0007674781372770667, -0.025155333802103996, -0.0016848113154992461, 0.023299502208828926, -0.02443738467991352, 0.011019852012395859, -0.03608713671565056, -0.015849079936742783, 0.003146110102534294, 0.0038200346753001213, 0.010403499007225037, -0.00040913824341259897, 0.005032421555370092, -0.01924918219447136, 0.02537207305431366, -0.007734892889857292, 0.0063768839463591576, -0.0014807713450863957, -0.01291632279753685, 0.029178561642766, -0.005547177977859974, 0.0019032441778108478, -0.02114565297961235, -0.0058181025087833405, 0.01966911368072033, -0.01613355055451393, 0.006268514320254326, -0.013282070867717266, -0.007281094323843718, 0.0014739983016625047, 0.004957917146384716, -0.023773619905114174, -0.002089504851028323, 0.0032477066852152348, 0.012875684536993504, 0.02592746913433075, 0.017135970294475555, 0.005452354438602924, 0.027485284954309464, -0.005625068675726652, -0.00409434549510479

In [22]:
# Let's see what it looks like (it will be a list of floating point numbers)
text = "I am PremKumar, you tutor and mentor for RAG and AI, I am an AI Architect"
vector = embedding_model.embed_query(text)
print(f"Vector length: {len(vector)}")
print(f"First 5 numbers: {vector}")

Vector length: 1536
First 5 numbers: [0.006043288856744766, -0.01451459527015686, 0.0021036129910498857, -0.02362467721104622, 0.011451145634055138, 0.019705068320035934, -0.030286675319075584, 0.008962929248809814, -0.029805084690451622, -0.015972742810845375, 0.014808900654315948, 0.004916233941912651, 0.01177220605313778, -0.0026872067246586084, -0.0013185205170884728, -0.02354441210627556, 0.021618051454424858, -0.005615208763629198, 0.0033728040289133787, 0.003040038514882326, -0.01314340066164732, 0.019878976047039032, -0.00887597631663084, -0.008173656649887562, -0.02690216712653637, -0.00793286133557558, 0.018915794789791107, -0.022648118436336517, 0.006942925974726677, -0.012387571856379509, 0.0008553243242204189, -0.0031654525082558393, 0.003936331253498793, -0.022313682362437248, 0.0009038178250193596, -0.0006007336778566241, 0.009424453601241112, 0.032213035970926285, 0.015972742810845375, 0.006845938973128796, 0.02801249921321869, -0.007912795059382915, -0.0028878692537546

Does the vector length change based on the size of the text?

text = "I am PremKumar, you tutor and mentor for RAG and AI"
If I change the text to "I am PremKumar, you tutor and mentor for RAG and AI" the vector length will change.

No, the vector length never changes based on the size of the text; it remains fixed regardless of whether you input a single keyword or a detailed paragraph. 

Embedding models are designed to map variable-length text into a static, fixed-dimensional space (for example, OpenAI’s text-embedding-ada-002 always outputs a vector of exactly 1,536 numbers). 

This uniformity is essential because vector databases and similarity algorithms (like Cosine Similarity) require every data point to have the exact same number of dimensions to mathematically compare how "close" or related they are to one another.

**NOTE: Fixed Dimensionality

Embedding models are neural networks with a fixed output layer. They are designed to map variable-length text into a fixed-size vector space.

For OpenAI's text-embedding-ada-002: The vector length is always 1536.

For OpenAI's text-embedding-3-large: The default is 3072 (though this specific model allows you to programmatically shorten it).

If you ran your code with the word "Hi" and then again with a 500-word essay, len(vector) would print 1536 both times (assuming the standard model).